include/zephyr/drivers/i2c/target/eeprom.h

```diff
@@ -21,6 +21,23 @@
  * @{
  */
 
+/**
+ * @typedef eeprom_target_changed_handler_t
+ * @brief Define the application callback handler function signature
+ *
+ * @param dev Pointer to the device structure for the driver instance.
+ */
+typedef void (*eeprom_target_changed_handler_t)(const struct device *dev);
```

Comment:
Please modify this to also take a `void *user_data` argument as described here: https://docs.zephyrproject.org/latest/develop/api/design_guidelines.html#using-callbacks

Reply:
So in addition to the dev argument? Sure, will change